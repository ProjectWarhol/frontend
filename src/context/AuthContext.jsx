import * as SecureStore from 'expo-secure-store'
import createDataContext from "./createDataContext"
import Api from '../api/api'
import { navigate } from '../navigationRef'

const authReducer = (state, action) => {
  switch (action.type){
    case 'add_error':
      return {...state, errorMessage: action.payload }
    case 'login':
      return {errorMessage: '', cookie: action.payload}
    case 'clear_error_message':
      return { ...state, errorMessage: '' }
    case 'logout':
      return { cookie: null, errorMessage: '' }
    default:
      return state
  }
}

const logout = dispatch => async () => {
  try { const cookie = await SecureStore.getItemAsync('cookie')
  const config = {headers:{
    'my.sid': cookie
  }}
  await Api.post('/users/logout', config)
  await SecureStore.deleteItemAsync('cookie')
  dispatch({ type: 'logout'})
  navigate('unAuthenticatedUser')
}catch(err){
  const cookie = await SecureStore.getItemAsync('cookie')
    await SecureStore.deleteItemAsync('cookie')
    dispatch({ type: 'logout'})
    navigate('unAuthenticatedUser')
}}

const tryLocalLogin = dispatch => async () => {
  const cookie = await SecureStore.getItemAsync('cookie')
if(cookie && isCookieValid()){
  dispatch({ type: 'login', payload: cookie})
  navigate('mainFlow')
}else{
  navigate('unAuthenticatedUser')
}
}

const isCookieValid = async(cookie) => {
  const config = {headers:{
    'my.sid': cookie
  }}
  await Api.get('/users/session', config)
  return true
}

const clearErrorMessage = dispatch => () => {
  dispatch({ type: 'clear_error_message' })
}

const login = (dispatch) => async ({ email, password }) => {
    try{
      const response = await Api.post('/users/login', { userCredential: email, password: password })
      let cookieArray = response["headers"]["set-cookie"]
      let cookieString = cookieArray.toString()
      const cookie = cookieString.substring(
        cookieString.indexOf('=') + 1,
        cookieString.indexOf(';')
      )
      await SecureStore.setItemAsync('cookie', cookie)
      dispatch({ type: 'login', payload: cookie})
      navigate('mainFlow')
    } catch (err){
      {dispatch({ type: 'add_error',  payload: 'Something went wrong' }, console.log(err))}
    }
  }

export const { Provider, Context } = createDataContext(
  authReducer,
  { login, logout, clearErrorMessage, tryLocalLogin },
  { cookie: null, errorMessage: '' }
)
