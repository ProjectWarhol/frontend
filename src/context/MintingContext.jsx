import createDataContext from "./createDataContext"
import Api from '../api/api'
import { navigate } from '../navigationRef'

const mintingReducer = (state, action) => {
  switch (action.type){
    case 'image':
      return {...state, image: action.payload}
    default:
      return state
  }
}

const takePicture = dispatch => async (camera) => {
  if(camera){
    const data = await camera.takePictureAsync(null)
    dispatch({type: 'image', payload: data.uri} )
    navigate('image')
  }
}

export const { Provider, Context } = createDataContext(
  mintingReducer,
  { takePicture },
  { image: null }
)
