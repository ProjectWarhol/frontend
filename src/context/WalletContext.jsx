import createDataContext from "./createDataContext"
import Api from '../api/api'
import { navigate } from '../navigationRef'
// import {createCustodialWallet, storeCustodialWallet, generateSeedPhrase} from '../wallet/CustodialWallet'

const walletReducer = (state, action) => {
  switch (action.type){
    case('createWallet'):
      return {...state }
    case 'signup':
      return {...state }
    case 'clear_error_message':
      return { ...state, errorMessage: '' }
    case 'add_error':
      return {...state, errorMessage: action.payload }
    default:
      return state
  }
}

const signup = (dispatch) => {
  return async ({ email, password, repeatedPassword, userName }) => {
    try{
      if (password === repeatedPassword)
      {const response = await Api.post('/users/createUser', { email: email, password: password, userName: userName })
      console.log(response.data)
      dispatch({ type: 'signup' })
      navigate('selectWallet')}
      else{dispatch({type: 'add_error', payload: 'Passwords need to match'})}
    } catch (err)
    {dispatch({ type: 'add_error', payload: 'Something went wrong' }, console.log(err))
    }
  }
}

const clearErrorMessage = dispatch => () => {
  dispatch({ type: 'clear_error_message' })
}

export const { Provider, Context } = createDataContext(
  walletReducer,
  { signup, clearErrorMessage },
  { errorMessage: '', email: '', password: '', userName: ''}
)
