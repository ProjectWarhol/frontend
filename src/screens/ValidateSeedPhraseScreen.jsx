import React, { useState, useContext, useEffect } from 'react'
import { StyleSheet, View, TextInput } from 'react-native'
import { Text, Button } from 'react-native-elements'
import { NavigationEvents } from 'react-navigation'
import { Context as WalletContext } from '../context/WalletContext'

const ValidateSeedPhraseScreen = ({navigation}) => {
    const { validateInput, clearErrorMessage, state } = useContext(WalletContext)
    const [seedPhrase, setSeedPhrase] = useState('')
    const testPhrase = 'blouse diet retreat cry sun badge return decide ski stick glory lazy'

    const checkSeedPhrase = () => {
     validateInput(seedPhrase, testPhrase)}

    return (
        <View style={styles.container}>
            <NavigationEvents
            onWillFocus={clearErrorMessage}
            />
            <NavigationEvents />
            <Text h3>Validate Your Seed Phrase</Text>
            <TextInput
                style={styles.input}
                editable
                multiline
                onChangeText={(text) => setSeedPhrase(text)}
                value={seedPhrase}
                placeholder="Your Validation Seed Phrase"
                autoCorrect={false}
                autoCapitalize='none'
                maxLength={200}
            />
            <Button
                title="Validate"
                onPress={() => checkSeedPhrase(seedPhrase, testPhrase)}
                style={styles.validateButton}
            />
            {state.errorMessage ? (<Text style={styles.errorMessage}>{state.errorMessage}</Text> ) : null}
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        marginBottom: 200,
        marginLeft: 25,
        marginRight: 25,
        marginTop: 50,
    },
    input: {
        height: 200,
        margin: 12,
        borderWidth: 1,
        padding: 10,
        fontSize: 20,
        borderRadius: 10
    },
    validateButton:{
      paddingBottom: 15
    }
})
export default ValidateSeedPhraseScreen
