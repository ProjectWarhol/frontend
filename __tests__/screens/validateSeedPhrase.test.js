import React, {useContext} from "react";
import renderer from 'react-test-renderer';
import 'react-native-gesture-handler/jestSetup';

import ValidateSeedPhraseScreen from '../../src/screens/ValidateSeedPhraseScreen';
import { Context as AuthContext, AuthProvider, validateInput } from '../../src/context/AuthContext';

jest.useFakeTimers()

jest.mock('@expo/vector-icons/build/vendor/react-native-vector-icons/lib/create-icon-set.js', () => {
  return () => '';
});

jest.mock('react-native-gesture-handler', () => { return {} });

jest.mock('react-navigation', () => ({
  withNavigation: ValidateSeedPhraseScreen => props => (
    <ValidateSeedPhraseScreen navigation={{ navigate: jest.fn() }} {...props} />
  ), NavigationEvents: 'mockNavigationEvents'
}));

test('renders correctly', async () => {
  const tree = await renderer.create(
    <AuthProvider value={null}>
      <ValidateSeedPhraseScreen />
    </AuthProvider>, {}).toJSON();
  expect(tree).toMatchSnapshot();
});


