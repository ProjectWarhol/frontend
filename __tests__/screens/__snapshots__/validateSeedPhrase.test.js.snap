// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "marginLeft": 25,
      "marginRight": 25,
    }
  }
>
  <mockNavigationEvents
    onWillFocus={[Function]}
  />
  <mockNavigationEvents />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginRight": "70%",
        "opacity": 1,
      }
    }
  >
    <Text />
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Go Back
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": "40%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#242424",
          "fontSize": 36,
        }
      }
    >
      Validate Your Seed Phrase
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      editable={true}
      maxLength={200}
      multiline={true}
      onChangeText={[Function]}
      placeholder="Your Validation Seed Phrase"
      rejectResponderTermination={true}
      style={
        Object {
          "borderRadius": 10,
          "borderWidth": 1,
          "fontSize": 20,
          "height": 200,
          "margin": 12,
          "padding": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 3,
          },
          undefined,
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "paddingBottom": 15,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Validate
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
